src/buildBrazeMessageComponent.tsx

```diff
@@ -1,15 +1,27 @@
 import React from 'react';
+import type { BrazeClickHandler, SubmitComponentEvent, TrackClick } from './utils/tracking';
+import { buildTrackClick, OphanComponentType } from './utils/tracking';
 
 interface HasComponentName {
     componentName: string;
 }
 
+interface HasClickTracking {
+    logButtonClickWithBraze: BrazeClickHandler;
+    submitComponentEvent: SubmitComponentEvent;
+}
+
+export interface HasTrackClick {
+    trackClick: TrackClick;
+}
+
 export type ComponentMapping<A> = {
     [key: string]: React.FC<A>;
 };
 
-export function buildBrazeMessageComponent<A extends HasComponentName>(
-    mappings: ComponentMapping<A>,
+export function buildBrazeMessageComponent<A extends HasComponentName & HasClickTracking>(
```

Comment:
I'm guessing the ampersand is either an intersection type? i.e A must implement both `HasComponentName` and `HasClickTracking`

Reply:
Yep that's right 👍  https://www.typescriptlang.org/docs/handbook/2/objects.html#intersection-types